### PR TITLE
[Deneb] Remove `BlindedBlobSidecar` and clean up block production/publishing flow

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description

- Delete `BlindedBlobSidecar`, `SignedBlindedBlobSidecar` and all related classes.
- Don't unblind blob sidecars in `BlockFactoryDeneb`
- No need of `BlindBlockUtilDeneb` so simplification there
- Remove all property tests

## Fixed Issue(s)
helps with block production flow for https://github.com/Consensys/teku/issues/7654

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockFactoryDeneb.java [hunk 1]
```diff
@@ -124,22 +115,10 @@
...
-    final List<SignedBlindedBlobSidecar> blindedBlobSidecars =
-        blindedBlockContainer
-            .toBlinded()
-            .flatMap(SignedBlindedBlockContainer::getSignedBlindedBlobSidecars)
-            .orElse(Collections.emptyList());
-    return spec.unblindSignedBlindedBlobSidecars(
-        slot, blindedBlobSidecars, operationSelector.createBlobSidecarsUnblinderSelector(slot));
-  }
-
+  @Deprecated
```
Comment (zilm13) [inline beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockFactoryDeneb.java]: I'd remove `@Deprecated` as it's a path we are going to use in the future

File: beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactoryTest.java [hunk 2]
```diff
@@ -521,24 +517,6 @@ void shouldCreateBlobSidecarsForBlockFromCachedPayloadResult() {
...
-    final CapturingBlobSidecarsUnblinder blobSidecarsUnblinder =
-        new CapturingBlobSidecarsUnblinder();
-
-    factory.createBlobSidecarsUnblinderSelector(slot).accept(blobSidecarsUnblinder);
-
-    assertThat(blobSidecarsUnblinder.blobsBundle)
-        .isCompletedWithValue(executionPayloadAndBlobsBundle.getBlobsBundle());
-  }
-
   private void prepareBlockProductionWithPayload(
```
Comment (zilm13) [inline beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactoryTest.java:520]: We will have the same flow here for blinded and unblinded? Why `shouldSetBlobsBundle_whenAcceptingTheBlobSidecarsUnblinderSelector` is no more needed?

**Review by zilm13 — APPROVED** [review]
LGTM, just few nits

File: beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockFactoryDeneb.java (hunk 1 above)
Comment (StefanBratanov) [reply beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockFactoryDeneb.java]: you are right, ta

File: beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactoryTest.java (hunk 2 above)
Comment (StefanBratanov) [reply beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactoryTest.java:520]: I basically deleted the method, not sure right now what would be the best way to do this, so will add test when working on the block production.

**Pull request merged**